Demo/src/main/java/com/braintreepayments/demo/PayPalNativeFragment.java

```diff
@@ -4,29 +4,99 @@
+        billingAgreementButton = view.findViewById(R.id.paypal_native_billing_agreement_button);
+        singlePaymentButton = view.findViewById(R.id.paypal_native_single_payment_button);
+
+        billingAgreementButton.setOnClickListener(this::launchBillingAgreement);
+        singlePaymentButton.setOnClickListener(this::launchSinglePayment);
+
+        DemoViewModel viewModel = new ViewModelProvider(getActivity()).get(DemoViewModel.class);
+        viewModel.getPayPalBrowserSwitchResult().observe(getViewLifecycleOwner(), this::handlePayPalBrowserSwitchResult);
+        return view;
     }
 
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    public void launchSinglePayment(View v) {
         getBraintreeClient(braintreeClient -> {
             payPalNativeClient = new PayPalNativeClient(braintreeClient);
             startPayPalNative();
         });
     }
 
+    public void launchBillingAgreement(View v) {
+        FragmentActivity activity = getActivity();
+        activity.setProgressBarIndeterminateVisibility(true);
+
+        getBraintreeClient((braintreeClient) -> {
+            if (braintreeClient == null) {
+                return;
+            }
+
+            payPalNativeClient = new PayPalNativeClient(braintreeClient);
+            dataCollector = new DataCollector(braintreeClient);
+
+            braintreeClient.getConfiguration((configuration, configError) -> {
+                if (getActivity().getIntent().getBooleanExtra(MainFragment.EXTRA_COLLECT_DEVICE_DATA, false)) {
+                    dataCollector.collectDeviceData(activity, (deviceData, dataCollectorError) -> this.deviceData = deviceData);
+                }
+                payPalNativeClient.tokenizePayPalAccount(activity, createPayPalNativeVaultRequest(activity), payPalError -> {
+                    if (payPalError != null) {
+                        handleError(payPalError);
+                    }
+                });
+            });
+        });
+    }
+
+    public void handlePayPalBrowserSwitchResult(BrowserSwitchResult browserSwitchResult) {
+        if (browserSwitchResult != null) {
+            payPalNativeClient.onActivityResumed(browserSwitchResult, (payPalAccountNonce, error) -> handlePayPalResult(payPalAccountNonce, error));
+        }
+    }
+
+    private void handlePayPalResult(PaymentMethodNonce paymentMethodNonce, Exception error) {
+        if (paymentMethodNonce != null) {
+            super.onPaymentMethodNonceCreated(paymentMethodNonce);
+
+            PayPalNativeFragmentDirections.ActionPayPalNativeFragmentToDisplayNonceFragment action =
+                    PayPalNativeFragmentDirections.actionPayPalNativeFragmentToDisplayNonceFragment(paymentMethodNonce);
+            action.setDeviceData(deviceData);
+
+            NavHostFragment.findNavController(this).navigate(action);
+        }
+    }
+
```

Comment:
Most of the modifications inside `PayPalNativeFragment` are a copy of whats inside `PayPalFragment`. The idea is to replicate the same behaviour for vault in both `PayPal` and `PayPalNative`